### ISSUE TITLE
allow setting token via environment variable for `spin cloud deploy`

### ISSUE THREAD

**Issue by rajatjindal** [fermyon/spin#1163]
it will be useful to support setting an env variable which can be consumed by `spin cloud deploy` to do the deployment.

**Comment by vdice** [comment]
Can you elaborate on the use case @rajatjindal?  Perhaps for scenarios where the `spin cloud login` action to Fermyon Cloud has been done separate from the current environment (i.e. the json config wouldn't be present locally at `spin cloud deploy` time?)

Question on precedence: If the env var is non-empty, I'm assuming its value should take precedence over the token stored in the json config, if exists, for the current deployment environment?

**Comment by rajatjindal** [comment]
> Can you elaborate on the use case @rajatjindal? Perhaps for scenarios where the `spin cloud login` action to Fermyon Cloud has been done separate from the current environment (i.e. the json config wouldn't be present locally at `spin cloud deploy` time?)
> 

yes, precisely that. if we have a cloud token, we should be able to use that directly without having to write it to a file in a specific format. 

> Question on precedence: If the env var is non-empty, I'm assuming its value should take precedence over the token stored in the json config, if exists, for the current deployment environment?

yes, I would think so too. 

on the other hand, if we want to introduce a flag `--token` instead of environment variable, that works too.

**Comment by lann** [comment]
You'd need an env var for server URL (and arguably "danger_accept_invalid_certs") too, wouldn't you? I wonder if we'd be better off just adding a `--token` opt to `spin cloud login` to update the credentials file.

**Comment by endocrimes** [comment]
I'd be a big proponent of adding env vars over _requiring_ we write config to a fixed file - esp for CI/CD Workflows where you probably wanna pull those from "some" kind of secret store
